### DISCUSSION TITLE
Work like "thefuck" and use AI to automatically correct previous command

### DISCUSSION THREAD

**Discussion by FunnySaltyFish** [TheR1D/shell_gpt#177] [Ideas]
[thefuck](https://github.com/nvbn/thefuck) is a command line tool which can automatically correct your last command. When something gets wrong, just type `fuck` and it will try to correct it.

Thefuck uses user-defined rules to do the correction, but **we can use AI now**! We can also provide some information to help AI give the correct command. The command might be like:
```bash
sgpt fuck
sgpt fuck "some extra information"

**Comment by xsoc** [comment]
You could potentially use something along the lines of
```
sgpt --shell "I made a mistake here somewhere, can you fix this command for me: `history -1`"
```
`history -1` will display the last command you executed.

Make an alias for it and pop it in your shell profile :)

**Reply by FunnySaltyFish** [reply]
Thanks for advice, the problems of your solution are:
- No other information could be provided
- `history -1` is not suitable for all terminals, in my case, both `cmd` and `git bash` don't support it (Windows 11)
- Only the command we wrote can be sent to AI, but the output mistake cannot.

**Reply by xsoc** [reply]
I've don't know much about it, but `doskey /history` might get you one step closer on Windows.
